generator/README.md

```diff
@@ -0,0 +1,95 @@
+# Custom Resources Generator
+
+Up until release v0.12 metallb was configured via metallb's Configmap.
+Deprecating the configmap impacts on the users' current configurations.
+To mitigate this, the MetalLB resource generator is an offline tool to
+convert the configmap into the new CRDs.
+
+## Using the generator
+
+### Prerequisites
+
+To use the generator, first create the container image:
+```
+inv build-cr-generator
+```
+### Procedure
+
+Run the container and redirect the output to a file:
+```
+docker run -v <PATH_TO_CONFIG>:/var/input <IMAGE_ID> -origin <CONFIGMAP_FILE_NAME> > metallb_resources.yaml
+```
+
+## Example
+
+For this MetalLB configmap configurations:
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    peers:
+    - my-asn: 64512
+      peer-asn: 64512
+      peer-address: 10.96.0.100
+    address-pools:
+    - name: my-ip-space
+      protocol: bgp
+      addresses:
+      - 198.51.100.0/24
+      bgp-advertisements:
+      - aggregation-length: 32
+        localpref: 100
+        communities:
+        - 64512:1234
+```
+
+The generator will output the following yaml:
+```
+# This was autogenerated by MetalLB's custom resource generator.
+apiVersion: metallb.io/v1beta2
+kind: BGPPeer
+metadata:
+  creationTimestamp: null
```

Comment:
no need to have this, can you remove it?

Reply:
will do.

Reply:
And this comes directly from the go stdlib:
```
// UnmarshalJSON implements the json.Unmarshaler interface.
// The time is expected to be a quoted string in RFC 3339 format.
func (t *Time) UnmarshalJSON(data []byte) error {
	// Ignore null, like in the main JSON package.
	if string(data) == "null" {
		return nil
	}
	// Fractional seconds are handled implicitly by Parse.
	var err error
	*t, err = Parse(`"`+RFC3339+`"`, string(data))
	return err
}
```
(note that we use k8s deserializer, which converts from go to json and from json to yaml).

Reply:
And this comes directly from the go stdlib:
```
// UnmarshalJSON implements the json.Unmarshaler interface.
// The time is expected to be a quoted string in RFC 3339 format.
func (t *Time) UnmarshalJSON(data []byte) error {
	// Ignore null, like in the main JSON package.
	if string(data) == "null" {
		return nil
	}
	// Fractional seconds are handled implicitly by Parse.
	var err error
	*t, err = Parse(`"`+RFC3339+`"`, string(data))
	return err
}
```
(note that we use k8s deserializer, which converts from go to json and from json to yaml).